src/Collections/LTreeCollection.php

```diff
@@ -24,6 +45,37 @@ public function toTree(bool $usingSort = true): LTreeNode
             $model->getLtreeParentColumn()
         );
 
-        return $builder->build($this, $usingSort);
+        return $builder->build($collection ?? $this, $usingSort);
+    }
+
+    private function isConsistency(LTreeModelInterface $model): bool
+    {
+        $paths = collect();
+
+        foreach ($this->items as $item) {
+            $paths = $paths->merge($item->getLtreePath());
+        }
+
+        return $paths
+            ->unique()
+            ->diff($this->pluck($model->getKeyName()))
+            ->isEmpty();
+    }
+
+    private function appendAncestors(LTreeModelInterface $model): void
+    {
+        $paths = $this->pluck($model->getLtreePathColumn())->toArray();
+        $ids = $this->pluck($model->getKeyName())->toArray();
+
+        /** @var Model $model */
+        $parents = $model::parentsOf($paths)
+            ->whereKeyNot($ids)
+            ->get();
+
+        foreach ($parents as $item) {
+            $this->add($item);
+        }
+
+        $this->sortBy($model->getKeyName());
```

Comment:
Why do we sort them there?

Reply:
Fixed: https://github.com/umbrellio/laravel-ltree/pull/31/commits/ccea7e2879c6edf99e1a07e3df4c1fae456b26b5